#VRML_SIM R2022b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/robots/gctronic/e-puck/protos/E-puck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/Floor.proto"
EXTERNPROTO "../protos/MazeWallEastWest.proto"
EXTERNPROTO "../protos/MazeWallNorthSouth.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.5773502691896257 0.5773502691896257 0.5773502691896257 2.0943245915045035
  position 0.0008002688621150799 0.00021245452539981222 1.7368842371772077
}
TexturedBackground {
}
TexturedBackgroundLight {
  luminosity 1.5
}
DEF epuck E-puck {
  translation 0.375 -0.375 -6.394982769581531e-05
  rotation 0 0 1 1.57
  controller "Lab5_Task1"
  supervisor TRUE
  turretSlot [
    Lidar {
      translation 0.01 0 0.01
      rotation 0 0 1 3.1415
      horizontalResolution 360
      fieldOfView 6.28319
      numberOfLayers 1
      minRange 0.04
      noise 0.01
    }
    DEF Front DistanceSensor {
      translation 0.037 0 0
      name "front distance sensor"
      lookupTable [
        0 0 0
        1 1 0.1
      ]
    }
    DEF Right DistanceSensor {
      translation 0 -0.037 0
      rotation 0 0 1 -1.5708
      name "right distance sensor"
      lookupTable [
        0 0 0
        1 1 0.1
      ]
    }
    DEF Rear DistanceSensor {
      translation -0.037 0 0
      rotation 0 0 1 3.1415
      name "rear distance sensor"
      lookupTable [
        0 0 0
        1 1 0.1
      ]
    }
    DEF Left DistanceSensor {
      translation 0 0.037 0
      rotation 0 0 1 1.5708
      name "left distance sensor"
      recognitionColors [
        0 0 0
      ]
      lookupTable [
        0 0 0
        1 1 0.1
      ]
    }
    InertialUnit {
      rotation -1 0 0 0
    }
    Camera {
      translation 0.037 0 0
      name "cameraFront"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
    Camera {
      translation 0 0.037 0
      rotation 0 0 1 1.5708
      name "cameraRight"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
    Camera {
      translation -0.037 0 0
      rotation 0 0 1 3.14
      name "cameraRear"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
    Camera {
      translation 0 -0.037 0
      rotation 0 0 1 -1.5708
      name "cameraLeft"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
  ]
}
DEF LandMarkCylinder Solid {
  translation 0.5 0.5 0.14925
  rotation 0 0 1 1.57
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 1
        radius 0.08
      }
    }
  ]
  name "Red Landmark"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 1 0 0
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.5
      radius 0.045
    }
  }
  recognitionColors [
    1 0 0
  ]
}
DEF LandMarkCylinder Solid {
  translation -0.5 0.5 0.14925
  rotation 0 0 1 1.57
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 1
        radius 0.08
      }
    }
  ]
  name "Yellow Landmark"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 1 1 0
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.5
      radius 0.045
    }
  }
  recognitionColors [
    1 1 0
  ]
}
DEF LandMarkCylinder Solid {
  translation -0.5 -0.5 0.14925
  rotation 0 0 1 1.57
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 1
        radius 0.08
      }
    }
  ]
  name "Green Landmark"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 0 1 0
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.5
      radius 0.045
    }
  }
  recognitionColors [
    0 1 0
  ]
}
DEF LandMarkCylinder Solid {
  translation 0.5 -0.5 0.14925
  rotation 0 0 1 1.57
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 1
        radius 0.08
      }
    }
  ]
  name "Blue Landmark"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 0 0 1
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.5
      radius 0.045
    }
  }
  locked TRUE
  recognitionColors [
    0 0 1
  ]
}
Floor {
  size 1 1
}
MazeWallNorthSouth {
  translation -0.375 0.5 0.025
  name "Cell 0 North Wall"
}
MazeWallNorthSouth {
  translation -0.375 0.25 0.025
  name "Cell 0 South Wall"
}
MazeWallEastWest {
  translation -0.5 0.375 0.025
  name "Cell 0 West Wall"
}
MazeWallNorthSouth {
  translation -0.125 0.5 0.025
  name "Cell 1 North Wall"
}
MazeWallNorthSouth {
  translation -0.125 0.25 0.025
  name "Cell 1 South Wall"
}
MazeWallNorthSouth {
  translation 0.125 0.5 0.025
  name "Cell 2 North Wall"
}
MazeWallNorthSouth {
  translation 0.375 0.5 0.025
  name "Cell 3 North Wall"
}
MazeWallEastWest {
  translation 0.5 0.375 0.025
  name "Cell 3 East Wall"
}
MazeWallEastWest {
  translation -0.5 0.125 0.025
  name "Cell 4 West Wall"
}
MazeWallEastWest {
  translation 0 0.125 0.025
  name "Cell 5 East Wall"
}
MazeWallEastWest {
  translation 0.25 0.125 0.025
  name "Cell 6 East Wall"
}
MazeWallEastWest {
  translation 0.5 0.125 0.025
  name "Cell 7 East Wall"
}
MazeWallEastWest {
  translation -0.25 -0.125 0.025
  name "Cell 8 East Wall"
}
MazeWallEastWest {
  translation -0.5 -0.125 0.025
  name "Cell 8 West Wall"
}
MazeWallNorthSouth {
  translation -0.125 -0.25 0.025
  name "Cell 9 South Wall"
}
MazeWallEastWest {
  translation 0.25 -0.125 0.025
  name "Cell 10 East Wall"
}
MazeWallNorthSouth {
  translation 0.125 -0.25 0.025
  name "Cell 10 South Wall"
}
MazeWallEastWest {
  translation 0.5 -0.125 0.025
  name "Cell 11 East Wall"
}
MazeWallNorthSouth {
  translation -0.375 -0.5 0.025
  name "Cell 12 South Wall"
}
MazeWallEastWest {
  translation -0.5 -0.375 0.025
  name "Cell 12 West Wall"
}
MazeWallNorthSouth {
  translation -0.125 -0.5 0.025
  name "Cell 13 South Wall"
}
MazeWallNorthSouth {
  translation 0.125 -0.5 0.025
  name "Cell 14 South Wall"
}
MazeWallEastWest {
  translation 0.5 -0.375 0.025
  name "Cell 15 East Wall"
}
MazeWallNorthSouth {
  translation 0.375 -0.5 0.025
  name "Cell 15 South Wall"
}
